src/ClientRuntime/Microsoft.Rest.ClientRuntime/ServiceClient.cs

```diff
@@ -253,7 +269,7 @@ protected static HttpClientHandler CreateRootHandler()
         /// <summary>
         /// Gets the HttpClient used for making HTTP requests.
         /// </summary>
-        public HttpClient HttpClient { get; protected set; }
+        public virtual HttpClient HttpClient { get; protected set; }
 
```

Comment:
@hovsepm 
As mentioned in the description, there are two ways to solve this problem
1) Provide constructor with HttpClient parameter
2) Make the HttpClient virtual, so classes that inherit from ServiceClient can override their own HttpClient.

#2 The sole purpose of making it virtual is to allow derived classes to override whatever was set by default, in this case a default HttpClient.
The Fabricam test itself is scenario where someone can override HttpClient that was created in Service client.
Let's talk more about your concern about potential bug. Could you provide a potential bug that I might have miss.
I think making it HttpClient property virtual is less risky and far less code churn than providing an overload.